### PR TITLE
Discovery settings

### PR DESCRIPTION
**Pull request by shift-eleven**
## Description

I added the new guild discovery endpoints, models and entities.

## Checklist

- [x] I discussed this PR with the maintainer(s) prior to opening it.
- [x] I read the [contributing guidelines](./.github/CONTRIBUTING.md).
- [ ] I tested the changes in this PR.

### CODE REVIEW

File: src/Disqord.Rest/ActionProperties/Modify/ModifyGuildDiscoveryMetadataActionProperties.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace Disqord.Rest.ActionProperties.Modify
```
Comment (AnotherZane) [inline src/Disqord.Rest/ActionProperties/Modify/ModifyGuildDiscoveryMetadataActionProperties.cs:3]: namespace Disqord

File: src/Disqord.Rest/Extensions/Entity/RestEntityExtensions.Guild.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 using System.IO;
 using System.Threading.Tasks;
 using Disqord.AuditLogs;
+using Disqord.Rest.ActionProperties.Modify;
```
Comment (AnotherZane) [inline src/Disqord.Rest/Extensions/Entity/RestEntityExtensions.Guild.cs:6]: Remove this after fixing namespace on the other file

File: src/Disqord.Rest/Extensions/RestClientExtensions.Guild.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@
 using Disqord.Collections;
 using Disqord.Http;
 using Disqord.Models;
+using Disqord.Rest.ActionProperties.Modify;
```
Comment (AnotherZane) [inline src/Disqord.Rest/Extensions/RestClientExtensions.Guild.cs:8]: Remove this after fixing namespace on the other file

File: src/Disqord.Rest/Extensions/RestClientExtensions.Guild.cs [hunk 4]
```diff
@@ -403,6 +404,34 @@ public static async Task<IGuildWidget> FetchWidgetAsync(this IRestClient client,
...
+            var model = await client.ApiClient.ModifyDiscoveryMetadataAsync(guildId, content, options);
+            return new TransientGuildDiscoveryMetadata(client, guildId, model);
+        }
+
+        public static Task CreateDiscoverySubcategoryAsync(this IRestClient client, Snowflake guildId, int categoryId, IRestRequestOptions options = null)
+            => client.ApiClient.AddDiscoverySubcategoryAsync(guildId, categoryId, options);
+
+        public static Task DeleteDiscoverySubcategoryAsync(this IRestClient client, Snowflake guildId, int categoryId, IRestRequestOptions options = null)
+            => client.ApiClient.DeleteDiscoverySubcategoryAsync(guildId, categoryId, options);
+
```
Comment (AnotherZane) [inline src/Disqord.Rest/Extensions/RestClientExtensions.Guild.cs:434]: Rename methods to (`Discovery` -> `GuildDiscovery`)

File: src/Disqord.Rest/Extensions/RestClientExtensions.Discovery.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        public static async Task<IReadOnlyList<IGuildDiscoveryCategory>> FetchDiscoveryCategoriesAsync(this IRestClient client, IRestRequestOptions options = null)
+        {
+            var models = await client.ApiClient.FetchDiscoveryCategoriesAsync(options);
+            return models.ToReadOnlyList(client, static (x, client) => new TransientGuildDiscoveryCategory(client, x));
+        }
+        public static async Task<bool> ValidateDiscoverySearchTermAsync(this IRestClient client, string term, IRestRequestOptions options = null)
+        {
+            var model = await client.ApiClient.ValidateDiscoverySearchTerm(term, options);
+            return model.Valid;
+        }
```
Comment (AnotherZane) [inline src/Disqord.Rest/Extensions/RestClientExtensions.Discovery.cs:19]: Rename methods to (`Discovery` -> `GuildDiscovery`)

File: src/Disqord.Rest.Api/Methods/RestApiClientExtensions.Guild.cs [hunk 6]
```diff
@@ -256,6 +256,30 @@ public static Task SyncIntegrationAsync(this IRestApiClient client, Snowflake gu
...
+        {
+            var route = Format(Route.Guild.CreateDiscoverySubcategory, guildId, categoryId);
+            return client.ExecuteAsync(route, null, options);
+        }
+
+        public static Task DeleteDiscoverySubcategoryAsync(this IRestApiClient client, Snowflake guildId, int categoryId, IRestRequestOptions options = null)
+        {
+            var route = Format(Route.Guild.DeleteDiscoverySubcategory, guildId, categoryId);
+            return client.ExecuteAsync(route, null, options);
+        }
```
Comment (AnotherZane) [inline src/Disqord.Rest.Api/Methods/RestApiClientExtensions.Guild.cs:281]: Rename methods to (`Discovery` -> `GuildDiscovery`)

File: src/Disqord.Rest.Api/Methods/RestApiClientExtensions.Discovery.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+            return client.ExecuteAsync<GuildDiscoveryCategoryJsonModel[]>(route, null, options);
+        }
+
+        public static Task<ValidateSearchTermJsonModel> ValidateDiscoverySearchTerm(this IRestApiClient client, string term, IRestRequestOptions options = null)
+        {
+            var queryParameters = new[] { new KeyValuePair<string, object>("term", term) };
+            var route = Format(Route.Discovery.ValidateSearchTerm, queryParameters);
+
+            return client.ExecuteAsync<ValidateSearchTermJsonModel>(route, null, options);
+        }
```
Comment (AnotherZane) [inline src/Disqord.Rest.Api/Methods/RestApiClientExtensions.Discovery.cs:21]: Rename methods to (`Discovery` -> `GuildDiscovery`)

**Review by AnotherZane — CHANGES_REQUESTED** [review]

File: src/Disqord.Rest/ActionProperties/Modify/ModifyGuildDiscoveryMetadataActionProperties.cs (hunk 1 above)
Comment (Quahu) [reply src/Disqord.Rest/ActionProperties/Modify/ModifyGuildDiscoveryMetadataActionProperties.cs:3]: `Disqord.Rest`*, the other action properties are wrong.

**Pull request closed without merging**